layers/core_checks/cc_ray_tracing.cpp

```diff
@@ -2097,3 +2097,389 @@ bool CoreChecks::ValidateDeferredOperation(VkDevice device, VkDeferredOperationK
+            vvl::range<VkDeviceAddress> scratch_address_range = scratch_buffer_state->DeviceAddressRange();
+            
+            if (!scratch_address_range.empty()) {
+                for (const auto& dst_addresses_buffer_state : dst_addresses_buffer_states) {
+                    const vvl::range<VkDeviceAddress> dst_addresses_address_range = dst_addresses_buffer_state->DeviceAddressRange();
+                    if (dst_addresses_address_range.intersects(scratch_address_range)) {
+                        const LogObjectList objlist(commandBuffer, dst_addresses_buffer_state->Handle(), scratch_buffer_state->Handle());
+                        skip |= LogError("VUID-vkCmdBuildClusterAccelerationStructureIndirectNV-dstAddressesArray-10455", objlist,
+                                         error_obj.location.dot(Field::pCommandInfos).dot(Field::dstAddressesArray),
+                                         "%s address range %s intersects with scratchData address range %s",
+                                         FormatHandle(dst_addresses_buffer_state->Handle()).c_str(),
+                                         string_range_hex(dst_addresses_address_range).c_str(),
+                                         string_range_hex(scratch_address_range).c_str());
+                    }
+                }
+            }
+        }
+    }
+
+    return skip;
+}
+
+bool CoreChecks::PreCallValidateGetClusterAccelerationStructureBuildSizesNV(VkDevice device,
+                                                                            const VkClusterAccelerationStructureInputInfoNV *pinfo,
+                                                                            VkAccelerationStructureBuildSizesInfoKHR *pSizeInfo,
+                                                                            const ErrorObject &error_obj) const {
+    bool skip = false;
+    if (!enabled_features.clusterAccelerationStructure) {
+        skip |= LogError("VUID-vkGetClusterAccelerationStructureBuildSizesNV-clusterAccelerationStructure-10438", device,
+                         error_obj.location, "clusterAccelerationStructures feature was not enabled.");
+    }
+
+    if (pinfo->opType == VK_CLUSTER_ACCELERATION_STRUCTURE_OP_TYPE_BUILD_TRIANGLE_CLUSTER_NV ||
+        pinfo->opType == VK_CLUSTER_ACCELERATION_STRUCTURE_OP_TYPE_BUILD_TRIANGLE_CLUSTER_TEMPLATE_NV) {
+        const VkClusterAccelerationStructureTriangleClusterInputNV *triangle_input =
+            reinterpret_cast<const VkClusterAccelerationStructureTriangleClusterInputNV *>(pinfo->opInput.pTriangleClusters);
+
+        skip |= ValidateClusterAccelerationStructureTriangleClusterInputNV(triangle_input, error_obj.location.dot(Field::input).dot(Field::opInput).dot(Field::pTriangleClusters));
+    }
+
+    return skip;
+}
+bool CoreChecks::ValidateClusterAccelerationStructureTriangleClusterInputNV(
+    const VkClusterAccelerationStructureTriangleClusterInputNV *input, const Location &input_loc) const {
```

Comment:
I realize I messed up the previous suggestion

```suggestion
    const VkClusterAccelerationStructureTriangleClusterInputNV& input, const Location &input_loc) const {
```

background - there are many parts of the API where a struct can be null and if we pass it deep into functions, it can be hard to know if we actually need to null check it or not, so we turn to a reference asap from the entrypoint when possible